infra/aws/terraform/management-account/s3.tf

```diff
@@ -0,0 +1,203 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+# Cost and Usage
+
+module "cur_reports_s3_bucket" {
+  providers = {
+    aws = aws.shared-services
+  }
+
+  source = "terraform-aws-modules/s3-bucket/aws"
+
+  bucket = "k8s-infra-cur-reports-bucket"
+
+  # S3 bucket-level Public Access Block configuration
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+
+  attach_deny_insecure_transport_policy = true
+  attach_policy                         = true
+  policy                                = data.aws_iam_policy_document.cur_reports_s3_bucket.json
+
+  # Note: Object Lock configuration can be enabled only on new buckets
+  # https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_object_lock_configuration
+  object_lock_enabled = true
+  object_lock_configuration = {
+    rule = {
+      default_retention = {
+        mode = "COMPLIANCE"
+        years = 3
+      }
+    }
+  }
```

Comment:
`s3_bucket_object_lock_configuration` is part of the module I'm using to create the bucket: https://github.com/terraform-aws-modules/terraform-aws-s3-bucket/blob/6fa011f63ff55992f40751bf24fdf35520b7b61d/main.tf#L324-L338

> This is a “[for but not with](https://en.wikipedia.org/wiki/For_but_not_with)” scenario in case we want to turn on object lock one day in the future - that way, we won't need to use an object lock token.

So we can just turn on the object locking mechanism with `object_lock_enabled = true` and remove the rule.